### ISSUE TITLE
Allow setting the domain from which the watchdog will send its emails

### ISSUE THREAD

**Issue by this-user** [mailcow/mailcow-dockerized#3720]
**Is your feature request related to a problem? Please describe.**

Currently, the watchdog script will use the very same hostname on which Mailcow installed for the sender address of the emails it sends. Usually, this will be something like `mail.<MYDOMAIN>`. This suffers from the drawback of the mails being more likely to be classified as spam, which is what is currently happening to me.

**Describe the solution you'd like**
Allow the user to set a custom domain for the sender address, e.g. the same domain used for actual mailboxes, in order to reduce spamminess. This works with my external email provider, and watchdog mails now land in the inbox instead of the spam folder.

A simple patch to that effect is provided below.

```patch
diff --git a/data/Dockerfiles/watchdog/watchdog.sh b/data/Dockerfiles/watchdog/watchdog.sh
index 01cd62fc..f3a33be7 100755
--- a/data/Dockerfiles/watchdog/watchdog.sh
+++ b/data/Dockerfiles/watchdog/watchdog.sh
@@ -98,6 +98,11 @@ function mail_error() {
   else
     SUBJECT="Watchdog ALERT: ${1}"
   fi
+  if [[ -z ${WATCHDOG_SENDER_DOMAIN} ]]; then
+    SENDER_DOMAIN=${MAILCOW_HOSTNAME}
+  else
+    SENDER_DOMAIN=${WATCHDOG_SENDER_DOMAIN}
+  fi
   IFS=',' read -r -a MAIL_RCPTS <<< "${WATCHDOG_NOTIFY_EMAIL}"
   for rcpt in "${MAIL_RCPTS[@]}"; do
     RCPT_DOMAIN=
@@ -116,7 +121,7 @@ function mail_error() {
       --body-plain="${BODY}" \
       --add-header="X-Priority: 1" \
       --to=${rcpt} \
-      --from="watchdog@${MAILCOW_HOSTNAME}" \
+      --from="watchdog@${SENDER_DOMAIN}" \
       --hello-host=${MAILCOW_HOSTNAME} \
       --ipv4
       #--server="${RCPT_MX}"
diff --git a/docker-compose.yml b/docker-compose.yml
index 52d1f714..6cd4ec25 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -422,6 +422,7 @@ services:
         - DBROOT=${DBROOT}
         - USE_WATCHDOG=${USE_WATCHDOG:-n}
         - WATCHDOG_NOTIFY_EMAIL=${WATCHDOG_NOTIFY_EMAIL}
+        - WATCHDOG_SENDER_DOMAIN=${WATCHDOG_SENDER_DOMAIN}
         - WATCHDOG_NOTIFY_BAN=${WATCHDOG_NOTIFY_BAN:-y}
         - WATCHDOG_EXTERNAL_CHECKS=${WATCHDOG_EXTERNAL_CHECKS:-n}
         - WATCHDOG_MYSQL_REPLICATION_CHECKS=${WATCHDOG_MYSQL_REPLICATION_CHECKS:-n}
diff --git a/generate_config.sh b/generate_config.sh
index 9af919e3..180b62be 100755
--- a/generate_config.sh
+++ b/generate_config.sh
@@ -249,6 +249,9 @@ ALLOW_ADMIN_EMAIL_LOGIN=n
 #WATCHDOG_NOTIFY_EMAIL=a@example.com,b@example.com,c@example.com
 #WATCHDOG_NOTIFY_EMAIL=
 
+# Set the domain from which the watchdog will send
+#WATCHDOG_SENDER_DOMAIN=
+
 # Notify about banned IP (includes whois lookup)
 WATCHDOG_NOTIFY_BAN=y

```

**Comment by andryyy** [comment]
Do you have a DMARC policy for the parent domain? These policies are inherited. You can add a policy for the FQDN mailcow listens on. 

Other than that, a flexible sender domain does not hurt. The variable should default to the MAILCOW_HOSTNAME.

**Comment by this-user** [comment]
I do have a DMARC policy, and I could adjust it, but the 'mail' subdomain isn't really supposed to be sending anything. So, to me it seems cleaner just being able to set the sender to the proper domain.

The sender does default to MAILCOW_HOSTNAME inside the watchdog.sh script if WATCHDOG_SENDER_DOMAIN is unset, but feel free to change that around however you like to do it.

**Comment by dragoangel** [comment]
@this-user the issue here is that watchdog is designed to not have access to dkim secret keys and work when other parts of mailcow - not. So it doesn't use postfix to send mail or rspamd to sign mail. So as described in help note: mail isn't signed by dkim. If you want put your watchdog email inside your main domain then it means that you main domain dmarc policy must reject only mail failed by spf and pass with failed dkim, which obviously isn't desired flow. With normal policy for main domain & additional dmarc policy for your mailcow.example.com subdomain you wouldn't have such bad impact. Do you have any idea how overcome this?

**Comment by this-user** [comment]
Mails with missing DKIM signature should still get through as long as SPF passes, and, in fact, do for me. So, sending from a subdomain without DKIM is basically the same as doing the same from the main domain. The only difference is that if you don't have a DMARC policy set for that subdomain, the mails are more likely to be classified as spam. And if you do add a DMARC entry with p=none, you are opening a way for a third party to exploit that. Either way, there is no advantage in not sending from the main domain. 

The only way to fully sidestep these issues would be adding a set of configuration options for sending through an external SMTP.

**Comment by dragoangel** [comment]
External SMTP is a way, as well as pushover integration or telegram bot or slack bot etc is possible way.

**Comment by dragoangel** [comment]
@andryyy what you think?

**Comment by andryyy** [comment]
Yes, it is a good idea. I will add this feature as well as the feature request in this thread. Just need a little bit more time.

**Comment by pittbull** [comment]
Will this be suggestion be implemented?

I would also like to set the sender address for consistency (and also because I'm getting NDR's when trying to deliver to the sub.domain.tld I have set up).

**Comment by normanu** [comment]
I was sending mails from Watchdog to a external email adres, they never arrived because of SPF / DMARC and the hostname not being in there (mail.domain.tld)
I removed the SPF and DMARC from my DNS for another reason, now suddenly I'm getting watchdog mails :)
